pwiz_tools/Skyline/Model/TransitionGroup.cs

```diff
@@ -743,15 +754,15 @@ protected override IList<ExplicitLoss> CreateState(ExplicitLoss[] mods)
                         else
                         {
                             allLosses = new List<TransitionLosses> { firstLosses };
-                            allLossMasses = new HashSet<double>();
-                            allLossMasses.Add(firstLosses.Mass);
+                            allLossMassCharges = new HashSet<Tuple<double, int>>();
+                            allLossMassCharges.Add(new Tuple<double, int>(firstLosses.Mass, firstLosses.TotalCharge));
                             firstLosses = null;
                         }
                     }
                     if (allLosses != null)
                         allLosses.Add(tranLosses);
-                    if (allLossMasses != null)
-                        allLossMasses.Add(tranLosses.Mass);
+                    if (allLossMassCharges != null)
+                        allLossMassCharges.Add(new Tuple<double, int>(tranLosses.Mass, tranLosses.TotalCharge));

```

Comment:
If your goal is to prevent this function from returning losses which have the same mass, then it would make more sense if you used "lossCharge" instead of "tranLosses.TotalCharge" (but, then, the behavior of this function would be incorrect).

Reply:
This is a very good point, Nick. I cleaned up the code to use a new struct called LossId to make the intention clearer and more correct. And I added a new test with just charged losses which failed before, because it turned out that the old code relied on the neutral loss being present two exclude the other two loss states.